### DISCUSSION TITLE
Ask for prompt/input during Live Display? EDIT: possible workaround in reply

### DISCUSSION THREAD

**Discussion by erikputar** [Textualize/rich#1791] [Q&A]
I have a dashboard, that roughly looks  like this:
![image](https://user-images.githubusercontent.com/89874073/147857098-85c09700-5e81-4353-b919-75e91bf2efc7.png)

I use it with Live Display to continuosly update some scraped data, using Panels and all of the other renderables of course.

I was wondering if it was possible to ask for prompt/input during a Live display.

Thank you in advance!

**_EDIT: apparently it is not possible (yet)! Checked with previous questions! Sorry!_**

**Comment by erikputar** [comment]
I follow up on my own question. I tried doing something with a script that writes to a file, while the dashboard script reades it (notice the "_command interface_" on the dashboard):
![rich_commands](https://user-images.githubusercontent.com/89874073/147888496-8bf45bae-a434-4541-97f7-39c9082f2238.gif)

Does anyone have a better idea?

**Reply by pitosalas** [reply]
how did you do that? Can you share the source or a sample that shows the approach?

**Comment by zuhataslan** [comment]
I was looking for a similar thing. I ended up using a non-blocking input reader and display every char in a panel as I press them. You could do a similar thing and based on input trigger functions if the input is a command. I used getch from getchlib which can get input in non-block mode:

This is a snippet:
```py

with Live(layout, refresh_per_second=10, transient=True, screen=True):
	while True:
		get_ch(text)
		pass
``
get_ch: 
```py

text = Text('')

def get_ch(text: Text) -> Text:
	while True:
		ch:str = getchlib.getkey(False, echo=False)
		if ch.isprintable():
			text.append(ch)
		elif ch == '\x7f':
			# If backspace, crop last ch
			text.right_crop()
		elif ch == '\n':
			# Do nothing right now
			pass
```

using layout:
```py
layout = make_layout()
layout['header'].update(Header())
layout['chat'].update(make_chat_panel())
layout['userlist'].update(make_userlist_panel())
layout['footer'].update(Panel(Align.left(text, vertical='top'), box=box.ROUNDED, title_align='left', title='Input:'))
```
the input is done in shown and updated in layout['footer']

Hope it helps

**Comment by dlax** [comment]
Similar to https://github.com/Textualize/rich/discussions/1641.

**Comment by attheodo** [comment]
@zuhataslan I am trying to achieve a similar behaviour but having a hard time. If I use a renderable timestamp on one of the Panels, using your pattern above I see that it doesn't update properly and looks like it's blocking. Is there a chance you can share a more comprehensive gist in case I am missing something?

**Reply by zuhataslan** [reply]
I was not able to find the code I'm referring to. It was a long time ago, and I was just testing things. But if you have a code snippet that I can run to see the problem _maybe_ I can suggest a solution

**Reply by the-dezeray** [reply]
which library were you using for getchlib
